test/dotnet-build.Tests/IncrementalTests.cs

```diff
@@ -100,20 +100,77 @@ public void TestNoVersionFile()
             CreateTestInstance();
             BuildProject().Should().HaveCompiledProject(MainProject, _appFrameworkFullName);
 
-            //delete version file
+            // delete version file
             var versionFile = Path.Combine(GetIntermediaryOutputPath(), ".SDKVersion");
             File.Exists(versionFile).Should().BeTrue();
             File.Delete(versionFile);
             File.Exists(versionFile).Should().BeFalse();
 
-            //assert build skipped due to no version file
+            // assert build skipped due to no version file
             BuildProject().Should().HaveSkippedProjectCompilation(MainProject, _appFrameworkFullName);
 
-            //the version file should have been regenerated during the build, even if compilation got skipped
+            // the version file should have been regenerated during the build, even if compilation got skipped
             File.Exists(versionFile).Should().BeTrue();
         }
 
         [Fact]
+        public void TestRebuildDeletedSource()
+        {
+            CreateTestInstance();
+            var buildResult = BuildProject();
+            buildResult.Should().HaveCompiledProject(MainProject, _appFrameworkFullName);
+
+            var sourceFile = Path.Combine(GetProjectDirectory(MainProject), "Program2.cs");
+            File.Delete(sourceFile);
+            Assert.False(File.Exists(sourceFile));
+
+            // second build; should get rebuilt since we deleted a source file
+            buildResult = BuildProject();
+            buildResult.Should().HaveCompiledProject(MainProject, _appFrameworkFullName);
```

Comment:
Add a third build to assert cache was updated and thus the build skipped (must have missed have lost that change during the merge).